packages/cli/src/ErrorReporting.ts

```diff
@@ -42,7 +42,7 @@ export const initErrorHandling = async () => {
 		if (originalException instanceof ExecutionBaseError && originalException.severity === 'warning')
 			return null;
 
-		if (originalException instanceof ReportableError) {
+		if (originalException instanceof ApplicationError) {
```

Comment:
Since `ExecutionBaseError` now extends `ApplicationError`, these checks could be unified, and we can most likely remove the `ExecutionBaseError` import all together here.

Reply:
Do you mean unifying like `if (level === 'warning' || severity === 'warning') return null;`?

If so, `severity` is present in `ExecutionBaseError` but absent in `ApplicationError`, so we'd still need to check for `originalException instanceof ExecutionBaseError`, meaning the check is moved but not unified. Might be best to clean this up once we do away with `severity` altogether?

```ts
if (originalException instanceof ApplicationError) {
	if (
		originalException instanceof ExecutionBaseError &&
		originalException.severity === 'warning'
	) {
		return null;
	}

	const { level, extra } = originalException;

	if (level === 'warning') return null;
	event.level = level;
	if (extra) event.extra = { ...event.extra, ...extra };
}
```

Reply:
We can do this in another PR.